examples/proc_GDS_pulse_capture.py

```diff
@@ -0,0 +1,91 @@
+
+atten_ratio = 101.35  # attenutation ratio
+show_all = False
+with psd.figlist_var() as fl:
+    for filename, nodename, label in [
+        ("240730_test_amp1_fin_pulse_calib.h5", "pulse_calib_1", "amplitude = 1"),
+        ("240730_test_amp0p1_fin_pulse_calib.h5", "pulse_calib_1", "amplitude = 0.1"),
+    ]:
+        d = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
+        beta = psd.ndshape(d).pop("t").alloc().rename("p_90", "desired_beta")
+        beta.setaxis("desired_beta", d.get_prop("desired_betas"))
+        t_v_beta = psd.ndshape(d).pop("t").alloc().rename("p_90", "At_p")
+        t_v_beta.setaxis(
+            "At_p",
+            list(
+                d.get_prop("acq_params")["amplitude"] * d.get_prop("set_p90s")[j]
+                for j in range(len(d["p_90"]))
+            ),
+        )
+        for j in range(len(d["p_90"])):
+            s = d["p_90", j].C
+            s.set_units("t", "s")
+            s *= atten_ratio
+            s /= sqrt(2)  # Vrms
+            s /= sqrt(50)  # V/sqrt(R) = sqrt(P)
+            if show_all:
+                fl.basename = r"programmed $\beta$ = %s" % str(
+                    s.get_prop("set_betas")[j]
+                )
+                fl.next(fl.basename)
+                # {{{ Plot raw capture
+                fl.plot(s, alpha=0.2, color="blue")
+                # }}}
+            # {{{ make analytic
+            s.ft("t", shift=True)
+            s = s["t":(0, None)]
+            s *= 2
+            s["t":0] *= 0.5
+            # }}}
+            if show_all:
+                s.ift("t")
+                # {{{ Plot analytic
+                fl.plot(abs(s), color="orange")
+                s.ft("t")
+                # }}}
+            # {{{ frequency filter
+            s["t":(0, 11e6)] *= 0
+            s["t":(24e6, None)] *= 0
+            # }}}
+            s.ift("t")
+            if show_all:
+                # {{{ plot and integrate 90 pulse
+                fl.plot(abs(s), color="red")
+                # }}}
+            beta90_int = abs(s).contiguous(lambda x: x > 0.01 * s.max())[0]
+            beta1 = abs(s["t":beta90_int]).integrate("t").data.item() * 1e6
+            beta["desired_beta", j] = beta1
+            t_v_beta["At_p", j] = beta1
+            if show_all:
+                plt.ylabel(r"$\sqrt{P_{pulse}}$")
+                plt.text(
+                    beta90_int[0] * 1e6 - 1,
+                    -1,
+                    r"$t_{90} \sqrt{P_{tx}} = %f s \sqrt{W}$" % beta1,
+                )
+                plt.axvline(beta90_int[0] * 1e6, ls=":", alpha=0.2)
+                plt.axvline(beta90_int[1] * 1e6, ls=":", alpha=0.2)
+        # {{{ make nddata for beta of 90 pulse and 180 pulse
+        fl.next(r"Measured $\beta$ vs programmed $\beta$")
+        fl.plot(beta, "o", label=label)
+        psd.gridandtick(plt.gca())
+        plt.ylabel(r"measured $\beta$ / $\mathrm{\mu s \sqrt{W}}$")
```

Comment:
I'm going to ignore from here to the end because I would end up repeating a lot of the comments from above. 

Here I understand what you're trying to do with this extra ND data that you've defined, but I still point out that it's not necessary.

All that you're trying to do, which makes sense, is to multiply your amplitude by your pulse length, so that you can put both the low power and the high power data on the same plot.

Reply:
I would point out that this plot that you are generating is not exactly any of the three plots that are part of our main goal right now. That is okay. 

It is the first plot turned on its side (with x and y interchanged).

I still hold that you want to add to the end of this a section where you flip the axis coordinates and the values of the data, and then determine your polynomial coefficients and your linear threshold in this code, so that you do not need to copy and paste entire calibration tables into the function that does the interpolation for you.

This is tied into a discussion that is on slack, or in the pr for the acquisition, or both.

Reply:
This has since been addressed